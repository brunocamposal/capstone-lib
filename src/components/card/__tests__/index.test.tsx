import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import Card from '../index'

describe("Render test", () => {
  it("Should render a card", () => {
    const tree = renderer
      .create(<Card imgUrl="Test url" description="Test Description" title="Test Title" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
describe('Teste Shallow Enzyme', () => {
  it('check attributes', () => {
    const wrapper = shallow(
      <Card
        title="Test title"
        description="test description"
        imgUrl="Test url"
      />
    )
    const src = wrapper.find("img").prop("src");
    expect(src).toBe("Test url")

    const description = wrapper.find('p').text()
    expect(description).toBe("test description")

    const title = wrapper.find('h2').text()
    expect(title).toBe("Test title")
  })
})



