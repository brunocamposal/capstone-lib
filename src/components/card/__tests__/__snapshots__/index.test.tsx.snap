// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot test Should render a card 1`] = `
<div
  className="sc-bdnylx fWlEos"
>
  <h2>
    Test Title
  </h2>
  <img
    src="Test url"
  />
  <p>
    Test Description
  </p>
</div>
`;
